lib/puppet/transaction/resource_harness.rb

```diff
@@ -144,17 +144,17 @@ def sync_if_needed(param, context)
 
       event = create_change_event(param, current_value, historical_value)
       event.status = "failure"
-      event.message = param.format("change from %s to %s failed: #{detail}",
+      event.message = param.format(_("change from %s to %s failed: "),
```

Comment:
@Magisus you are on the right track. We can't just include `detail` in the string that's passed to `param.format` because it gets `[redacted]`. So it has to be appended later as you're doing...except we have to cast it as a string to do so, because the `+` operator won't do it for us, like string interpolation does. This is a bit of ruby being inconsistently helpful:

```
[5] pry(#<Puppet::Transaction::ResourceHarness>)> "foo" + detail
TypeError: no implicit conversion of StandardError into String
from (pry):5:in `+'
[6] pry(#<Puppet::Transaction::ResourceHarness>)> "foo #{detail}"
=> "foo StandardError"
[7] pry(#<Puppet::Transaction::ResourceHarness>)> "foo " + detail.to_s
=> "foo StandardError"
```
 i.e.
```ruby
      event.message = param.format(_("change from %s to %s failed: "),
                                   param.is_to_s(current_value),
                                   param.should_to_s(param.should)) + detail.to_s
```